### PR TITLE
bank: execute simple bank operations

### LINKED ISSUE #13: bank: execute simple bank operations
User should be able to execute simple bank operations such as deposit and withdraw. 

For example: 
balance = 0, deposit = 10, result => account with amount of 10 
balance = 20, withdraw = 5, result => account with amount of 15 
balance = 15, withdraw = 16, result => error message "There is not enough amount in your bank account to withdraw 16"
balance = 0, deposit = -10, result => error message "Deposit with negative amount is not allowed" 

Check for valid Money Amounts: "12,32", "12.32", "12", "-12", "23.445"

### PR DESCRIPTION
**Pull request by martinmilev**
Deposit and withdraw functionality implemented. Database with frontend app.
Fixes #13

### CODE REVIEW

File: frontend/app/app.js [hunk 1]
```diff
@@ -24,3 +24,19 @@ app.controller('TransactionsHistoryCtrl', function($scope, $http) {
       $scope.transactions = response.data;
     });
 });
+
+app.controller('HomePageCtrl', function($scope, $http) {
+  $scope.account = {};
+  $http.get("/v1/")
```
Comment (mgenov) [inline frontend/app/app.js]: It should get some resource, not just "/v1/".

Resources are: 
 - "/v1/accountbalance"
 - "/v1/useraccount"

File: frontend/app/view/index_page.view.html [hunk 2]
```diff
@@ -1 +1,96 @@
...
+                        <button type='submit' class="btn btn-success" ng-click="submitForm()">Execute Transaction</button>
+                    </div>
+                </div>
+            </form>
+        </li>
+    </ul>
+</div>
+
+
+<!--<div class="container col-md-8 col-md-offset-1" ng-controller="HomePageCtrl">-->
```
Comment (mgenov) [inline frontend/app/view/index_page.view.html]: The commented code need to be removed.

File: frontend/index.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline frontend/index.html]: I don't get what is the goal of this page.

File: src/main/java/com/clouway/bank/persistence/DBProvider.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.google.inject.Provider;
+import com.mongodb.MongoClient;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoDatabase;
+
+/**
+ * @author Martin Milev <martinmariusmilev@gmail.com>
+ */
+
+public class DBProvider implements Provider<MongoDatabase> {
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/persistence/DBProvider.java]: You can pass MongoClient to the constructor, so the Main will be able to pass it as parameter.

File: src/main/java/com/clouway/bank/persistence/PersistentAccountRepository.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  private Provider<MongoDatabase> db;
+
+  @Inject
+  public PersistentAccountRepository(Provider<MongoDatabase> db) {
+    this.db = db;
+  }
+
+  @Override
+  public Account register(String name, Double balance) {
+    MongoCollection<Document> collection = db.get().getCollection("accounts");
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/persistence/PersistentAccountRepository.java]: You can remove duplication of db.get().getCollection("accounts"); by wrapping the invocation into an accounts() method.

So the code will look: accounts().insertOne(document);

File: src/test/java/com/clouway/bank/persistence/PersistentAccountRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+//import java.util.Optional;
+//
+//import static junit.framework.TestCase.assertTrue;
+//import static org.hamcrest.MatcherAssert.assertThat;
+//import static org.hamcrest.core.Is.is;
+//
+///**
+// * @author Martin Milev <martinmariusmilev@gmail.com>
+// */
+//public class PersistentAccountRepositoryTest {
```
Comment (mgenov) [inline src/test/java/com/clouway/bank/persistence/PersistentAccountRepositoryTest.java]: Doesn't look correct to be commented.

Comment (martinmilev) [inline src/test/java/com/clouway/bank/persistence/PersistentAccountRepositoryTest.java]: Fixed.

File: src/main/java/com/clouway/bank/persistence/PersistentAccountRepository.java (hunk 5 above)
Comment (martinmilev) [inline src/main/java/com/clouway/bank/persistence/PersistentAccountRepository.java]: done.

File: frontend/index.html (hunk 3 above)
Comment (martinmilev) [inline frontend/index.html]: deleted

File: frontend/app/view/index_page.view.html (hunk 2 above)
Comment (martinmilev) [inline frontend/app/view/index_page.view.html]: fixed.

File: src/main/java/com/clouway/bank/persistence/DBProvider.java (hunk 4 above)
Comment (martinmilev) [inline src/main/java/com/clouway/bank/persistence/DBProvider.java]: Could not figure out how to pass it from the main.

**Comment by martinmilev** [comment]
Ready for review.

File: pom.xml [hunk 7]
```diff
@@ -103,22 +104,22 @@
...
-            <goals>
-              <goal>exec</goal>
-            </goals>
-            <configuration>
-              <executable>frontend/bundle.sh</executable>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
+      <!--<plugin>-->
```
Comment (mgenov) [inline pom.xml]: The commented code need to be removed.

Comment (martinmilev) [inline pom.xml]: fixed.

File: src/main/java/com/clouway/bank/core/AccountRepository.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.bank.core;
+
+import java.util.Optional;
+
+/**
+ * @author Martin Milev <martinmariusmilev@gmail.com>
+ */
+public interface AccountRepository {
+  Account register(String pesho, Double v);
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/core/AccountRepository.java]:
```java
Account register(String name, Double initialBalance);
```

File: src/test/java/com/clouway/bank/adapter/http/HomePageServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+  private HomePageService homePageService = new HomePageService(userRepository, userSecurity);
+
+  @Test
+  public void happyPath() {
+    final Optional<Account> possibleAccount = Optional.of(new Account("id", "A", 1d));
+
+    context.checking(new Expectations() {{
+      oneOf(userSecurity).currentUser();
+      will(returnValue(new User(with(any(String.class)))));
```
Comment (mgenov) [inline src/test/java/com/clouway/bank/adapter/http/HomePageServiceTest.java]:
```java
oneOf(userSecurity).currentUser();
will(returnValue(new User("::any user id::"));

oneOf(userReposiutory).findUserAccount("::any user id::");
will(returnValue(Optional.of(account)));

```

File: src/test/java/com/clouway/bank/adapter/persistence/DatastoreRule.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+ */
+public class DatastoreRule extends ExternalResource {
+  private String collection = "";
+
+  public DatastoreRule(String collection) {
+    this.collection = collection;
+  }
+
+  public Provider<MongoDatabase> getDatabase() {
+    return new Provider<MongoDatabase>() {
```
Comment (mgenov) [inline src/test/java/com/clouway/bank/adapter/persistence/DatastoreRule.java]: Could be simplified with: 

```java
return Providers.of(new MongoClient("localhost", 27017).getDatabase("testBankApp"));
```

File: src/test/java/com/clouway/bank/adapter/persistence/PersistentAccountRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    assertTrue(!actual.isPresent());
+  }
+
+  @Test
+  public void updateAccountBalance() throws Exception {
+    Account account = accountRepository.register("A", 10d);
+    Double expected = 5d;
+    accountRepository.update(account.id, expected);
+    Double actual = accountRepository.findUserAccount(account.id).get().balance;
+    assertThat(actual, is(expected));
```
Comment (mgenov) [inline src/test/java/com/clouway/bank/adapter/persistence/PersistentAccountRepositoryTest.java]: assertThat(balance, is(5d)); is much more readable then: assertThat(actual, is(expected));

File: src/main/java/com/clouway/bank/core/AccountRepository.java (hunk 8 above)
Comment (martinmilev) [inline src/main/java/com/clouway/bank/core/AccountRepository.java]: fixed.

File: src/test/java/com/clouway/bank/adapter/http/HomePageServiceTest.java (hunk 9 above)
Comment (martinmilev) [inline src/test/java/com/clouway/bank/adapter/http/HomePageServiceTest.java]: fixed.

File: src/test/java/com/clouway/bank/adapter/persistence/DatastoreRule.java (hunk 10 above)
Comment (martinmilev) [inline src/test/java/com/clouway/bank/adapter/persistence/DatastoreRule.java]: fixed.

File: src/test/java/com/clouway/bank/adapter/persistence/PersistentAccountRepositoryTest.java (hunk 11 above)
Comment (martinmilev) [inline src/test/java/com/clouway/bank/adapter/persistence/PersistentAccountRepositoryTest.java]: fixed.

**Pull request merged**